lib/Graph/Context.cpp

```diff
@@ -45,6 +46,12 @@ void Context::clear() {
   map_.clear();
 }
 
+Tensor *Context::allocate(Placeholder *P) {
+  Tensor *T = new Tensor(P->getType());
```

Comment:
do we need to make sure that there are no existing tensors backing \p P?

Reply:
Yes, you are right. I will fix this.